built_in_tasks/target_tracking_task.py

```diff
@@ -0,0 +1,494 @@
+'''
+A generic target tracking task
+'''
+import numpy as np
+import time
+import os
+import math
+import traceback
+import random
+from collections import OrderedDict
+
+from riglib.experiment import traits, Sequence, FSMTable, StateTransitions
+from riglib.stereo_opengl import ik
+from riglib import plants
+
+from riglib.stereo_opengl.window import Window
+from .target_graphics import *
+
+## Plants
+# List of possible "plants" that a subject could control either during manual or brain control
+cursor = plants.CursorPlant()
+shoulder_anchor = np.array([2., 0., -15])
+chain_kwargs = dict(link_radii=.6, joint_radii=0.6, joint_colors=(181/256., 116/256., 96/256., 1), link_colors=(181/256., 116/256., 96/256., 1))
+chain_20_20_endpt = plants.EndptControlled2LArm(link_lengths=[20, 20], base_loc=shoulder_anchor, **chain_kwargs)
+chain_20_20 = plants.RobotArmGen2D(link_lengths=[20, 20], base_loc=shoulder_anchor, **chain_kwargs)
+
+plantlist = dict(
+    cursor=cursor,
+    chain_20_20=chain_20_20,
+    chain_20_20_endpt=chain_20_20_endpt)
+
+class TargetTracking(Sequence):
+    '''
+    This is a generic cued target tracking skeleton, to form as a common ancestor to the most
+    common type of motor control tracking tasks.
+    '''
+    status = dict(
+        wait = dict(start_trial="target"),
+        target = dict(success="reward", timeout="wait"),
+        reward = dict(reward_end="wait", stoppable=False, end_state=True)
+    )
```

Comment:
there's a (very minor) issue of the way the number of trials are calculated in bmi3d, it will skip trials that don't end in reward, since this is the only state with `end_state=True`. you would have to add a `timeout` state or something similar, also with `end_state=True`, if you want those to be counted as "trials" in the GUI. not sure it really matters. in analysis we can define trials by whatever we want